volume-providers/vpc/vpcclient/vpcvolume/util.go

```diff
@@ -0,0 +1,36 @@
+/*******************************************************************************
+ * IBM Confidential
+ * OCO Source Materials
+ * IBM Cloud Container Service, 5737-D43
+ * (C) Copyright IBM Corp. 2018, 2019 All Rights Reserved.
+ * The source code for this program is not  published or otherwise divested of
+ * its trade secrets, irrespective of what has been deposited with
+ * the U.S. Copyright Office.
+ ******************************************************************************/
+
+package vpcvolume
+
+import (
+	"fmt"
+	"log"
+	"regexp"
+	"runtime"
+	"time"
+)
+
+// Get execution time of a function
+func TimeTrack(start time.Time) {
+	elapsed := time.Since(start)
+
+	// Skip this function, and fetch the PC and file for its parent.
+	pc, _, _, _ := runtime.Caller(1)
+
+	// Retrieve a function object this functions parent.
+	funcObj := runtime.FuncForPC(pc)
+
+	// Regex to extract just the function name (and not the module path).
+	runtimeFunc := regexp.MustCompile(`^.*\.(.*)$`)
+	name := runtimeFunc.ReplaceAllString(funcObj.Name(), "$1")
+
+	log.Println(fmt.Sprintf("%s took %s", name, elapsed))
```

Comment:
Can we use the caller`s zap logger here? We need to pass `logger ` to TimeTrack() method .
Also , I'm in favour of passing `function Name ` as a parameter to this method. Finding the the function from `runtime`  dynamically is quite expensive . This method will be used by  lot of method . So it needs to be optimised

Reply:
Addressed